### ISSUE TITLE
Building Against netbsd-curses?

### ISSUE THREAD

**Issue by githububub** [martanne/dvtm#42]
Building against netbsd-curses appears to build successfully, but launching dvtm 0.15 as `dvtm` or `DVTM_TERM=rxvt dvtm` or `DVTM_TERM=rxvt-unicode dvtm` or `DVTM_TERM=screen dvtm` launches a terminal window that cannot accept any input and must be killed to close. 

>==> Starting build()...
cleaning
cleaning
dvtm build options:
CC dvtm.c
CC vt.c
CFLAGS   = -march=native -O2 -pipe -fstack-protector-all -fomit-frame-pointer -std=c99 -I. -DVERSION="0.15" -DNDEBUG -D_FORTIFY_SOURCE=2 -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_XOPEN_SOURCE_EXTENDED
LDFLAGS  = -Wl,-z,now -Wl,-O1,--sort-common,--as-needed,-z,relro -lc -lutil -lncursesw
CC       = cc
vt.c: In function ‘vt_forkpty’:
vt.c:1636:4: warning: ignoring return value of ‘chdir’, declared with attribute warn_unused_result [-Wunused-result]
    chdir(cwd);
    ^~~~~~~~~~
CC -o dvtm
==> Entering fakeroot environment...
==> Starting package()...
stripping executable
installing executable file to /tmp/dvtm/pkg/dvtm/usr/bin
installing manual page to /tmp/dvtm/pkg/dvtm/usr/share/man/man1
installing terminfo description
2 entries and 0 aliases written to dvtm.info.cdb
==> Tidying install...
  -> Removing doc files...
  -> Removing libtool files...
  -> Removing static library files...
  -> Stripping unneeded symbols from binaries and libraries...
  -> Compressing man and info pages...
==> Checking for packaging issue...
==> Creating package "dvtm"...
  -> Generating .PKGINFO file...
  -> Generating .BUILDINFO file...
  -> Generating .MTREE file...
  -> Compressing package...
  100 %        32.1 KiB / 100.0 KiB = 0.321                                    
==> Leaving fakeroot environment.
==> Finished making: dvtm 0.15-2 (Sat Nov 12 16:00:34 UTC 2016)

Is dvtm.info.cdb supposed to be installed to TERMINFO? I notice the https://www.archlinux.org/packages/community/x86_64/dvtm/files/ stock Arch package as dvtm and dvtm-256color installed to `/usr/share/terminfo/d` but there are no such databases installed when building against netbsd-curses. All that exists are dvtm.info and dvtm.info.cdb in the source directory.

Built with `LDFLAGS="-lcurses -lterminfo"` per netbsd-curses recommendations...

>==> Starting build()...
cleaning
cleaning
dvtm build options:
CC dvtm.c
CFLAGS   = -march=native -O2 -pipe -fstack-protector-all -fomit-frame-pointer -std=c99 -I. -DVERSION="0.15" -DNDEBUG -D_FORTIFY_SOURCE=2 -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_XOPEN_SOURCE_EXTENDED
CC vt.c
LDFLAGS  = -Wl,-z,now -Wl,-O1,--sort-common,--as-needed,-z,relro -lcurses -lterminfo
CC       = cc
vt.c: In function ‘vt_forkpty’:
vt.c:1636:4: warning: ignoring return value of ‘chdir’, declared with attribute warn_unused_result [-Wunused-result]
    chdir(cwd);
    ^~~~~~~~~~
CC -o dvtm
vt.o: In function `vt_forkpty':
vt.c:(.text+0x134e): undefined reference to `forkpty'
collect2: error: ld returned 1 exit status
make: *** [Makefile:25: dvtm] Error 1
==> ERROR: A failure occurred in build().
    Aborting...

Built with `LDFLAGS="-lncurses"` per dvtm build recipe of Sabotage Linux....
                                                                                    
>==> Starting build()...
cleaning
cleaning
dvtm build options:
CC dvtm.c
CC vt.c
CFLAGS   = -march=native -O2 -pipe -fstack-protector-all -fomit-frame-pointer -std=c99 -I. -DVERSION="0.15" -DNDEBUG -D_FORTIFY_SOURCE=2 -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_XOPEN_SOURCE_EXTENDED
LDFLAGS  = -Wl,-z,now -Wl,-O1,--sort-common,--as-needed,-z,relro -lncurses
CC       = cc
vt.c: In function ‘vt_forkpty’:
vt.c:1636:4: warning: ignoring return value of ‘chdir’, declared with attribute warn_unused_result [-Wunused-result]
    chdir(cwd);
    ^~~~~~~~~~
CC -o dvtm
vt.o: In function `vt_forkpty':
vt.c:(.text+0x134e): undefined reference to `forkpty'
collect2: error: ld returned 1 exit status
make: *** [Makefile:25: dvtm] Error 1
==> ERROR: A failure occurred in build().
    Aborting...

**Comment by rofl0r** [comment]
i wasn't aware that dvtm requires an own terminfo database entry.
do i conclude correctly that dvm compiles a terminfo description with tic and installs that into a systemwide directory, assuming it will be automatically picked up ?

**Comment by josuah** [comment]
@rofl0r Yes, it [does install](https://github.com/martanne/dvtm#warning-terminal-is-not-fully-functional) its own terminfo with `tic`, if this is what you meant.

From the [Makefile](https://github.com/martanne/dvtm/blob/master/Makefile#L59):

```
tic -s dvtm.info
```

**Comment by rofl0r** [comment]
i had time to look at this issue in-depth and documented my findings here: https://github.com/sabotage-linux/netbsd-curses/issues/19 . as a temporary workaround or just to test that dvtm works theoretically as expected, @githububub could rename the generated dvtm.info.cdb to $HOME/.terminfo.cdb, and then we can find a satisfactory permanent solution.

**Comment by githububub** [comment]
dvtm.info.cdb to $HOME/.terminfo.cdb and dvtm.info.cdb to $(PREFIX)/share/terminfo.cdb both do not resolve the no-op terminal window issue.

**Comment by rofl0r** [comment]
i've tried it in xterm and there it worked.
i started dvtm like this to find out where/if it picks up the terminfo files:
```
strace -f dvtm 2>&1 | grep open
```
(failure to find the file has an ENOENT return value)
once it could find the .cdb with the dvtm entries in it, it worked correctly.

btw you can exit the no-op dvtm with CTRL-D

**Comment by rofl0r** [comment]
i've now added dvtm terminfos to netbsd-curses' terminfo db, so this should be a non-issue in the future (unless there will be major changes to the terminfo of dvtm in the future).
https://github.com/sabotage-linux/netbsd-curses/commit/396a9d047c6c8b9ae2aefa7ad3aa1641f511234d
